scripts/outline.js

```diff
@@ -1,61 +1,53 @@
+ */
+function createOutline(elementType, targetId) {
+    // Get all the headings in the document
+    let headings = document.querySelectorAll(elementType);
 
-  // Create an empty array to store the outline items
-  let outline = [];
+    // Create an empty array to store the outline items
+    let outline = [];
 
-  // Loop through the headings and extract the text and level
-  for (let heading of headings) {
-    let text = heading.textContent;
-    let level = parseInt(heading.tagName.slice(1));
-    level -= 2;
-    // Get the index of the heading in the headings array
-    let index = Array.prototype.indexOf.call(headings, heading);
-    // Generate an id based on the index
-    let id = "heading-" + index;
-    // Set the id attribute of the heading element
-    heading.setAttribute("id", id);
-    // Push an object with the text, level and id to the outline array
-    outline.push({text, level, id});
-  }
-
-  // Create a function to display the outline as a nested list
-  function displayOutline(outline) {
-    // Create an empty string to store the html output
-    let html = "";
-    // Keep track of the current level and the previous level
-    let currentLevel = 0;
-    let previousLevel = 0;
-    // Loop through the outline items
-    for (let item of outline) {
-      // Get the text and level of the item
-      let text = item.text;
-      let level = item.level;
-      // Update the current level
-      currentLevel = level;
-      // If the current level is greater than the previous level, start a new list
-      if (currentLevel > previousLevel) {
-        html += "<ul>";
-      }
-      // If the current level is less than the previous level, close the list
-      if (currentLevel < previousLevel) {
-        html += "</ul>".repeat(previousLevel - currentLevel);
-      }
-      // Get the id of the corresponding html element
-      let id = item.id;
-      // Add a list item with a hyperlink to the html element
-      html += "<li><a href='#" + id + "'>" + text + "</a></li>";
-      // Update the previous level
-      previousLevel = currentLevel;
-    }
-    // Close any remaining lists
-    html += "</ul>".repeat(currentLevel);
-    // Return the html output
-    return html;
-  }
+    // Process each heading and add to the outline array
+    headings.forEach((heading, idx) => {
+        text = heading.textContent;
+        id = `${elementType}-${idx}`;
+        // Set an ID for each heading
+        heading.setAttribute("id", id);
+        outline.push({text, id});
+    });
+    
+    
+    // Create a link to each heading inside an unordered list
+    let contentList = document.createElement("ul");
+    outline.forEach( o => { 
+        let li = document.createElement("li");
+        let a = document.createElement("a");
+        a.setAttribute("href", `#${o.id}`);
+        // Set title attribute, to help with long text
+        a.setAttribute("title", o.text.trim());
```

Comment:
A tooltip appears on hover of the link since often the text gets cut off. Tooltip shows all the text. Remove if not desired. 
![image](https://github.com/dbs-sticky/OnlyRSS/assets/746276/a2bbd2c3-63d7-4242-adc6-a2267db66fc5)

Reply:
ha, that was on my ToDo list!